alcs-frontend/src/app/features/home/gis/gis-subtasks.component.ts

```diff
@@ -66,14 +59,50 @@ export class GisSubtasksComponent implements OnInit {
     });
   }
 
+  private getMappedApplicationSubtasks(subtasks: ApplicationSubtaskWithApplicationDto[]) {
+    return subtasks
+      .filter((subtask) => subtask.application)
+      .map((subtask) => {
+        const statusDto = this.statuses.find((status) => status.code === subtask.application.status);
+        const userDto = this.gisUsers.find((user) => user.uuid === subtask.assignee);
+        return {
+          ...subtask,
+          assignee: userDto ? userDto.name : undefined,
+          application: {
+            ...subtask.application,
+            status: statusDto!.label,
+          },
+        };
+      });
+  }
+
+  private getMappedReconsiderationSubtasks(subtasks: ApplicationSubtaskWithApplicationDto[]) {
+    return subtasks
+      .filter((subtask) => subtask.reconsideration)
+      .map((subtask) => {
+        const statusDto = this.statuses.find((status) => status.code === subtask.reconsideration.card.status.code);
+        const userDto = this.gisUsers.find((user) => user.uuid === subtask.assignee);
+        return {
+          ...subtask,
+          assignee: userDto ? userDto.name : undefined,
+          application: {
+            ...(subtask.reconsideration.application as any),
```

Comment:
why do you need the any here?

Reply:
it is different from application in application subtask. The application subtask has other fields that redundant for reconsideration application

Reply:
If it only has extra fields you shouldn't need to cast as its duck typing. Is there a way to properly map it?

Reply:
Unfortunately, it is the other way around, it does not have the active days and other application-specific fields. In theory it is possible to rework the SubtaskDto to have only the required list of fields and not everything from the application.